index.src.html

```diff
@@ -2394,14 +2420,692 @@
+    [=ReadableStream/disturbed=] or [=ReadableStream/locked=], return `false`.
+3. If |data| is of type {{BufferSource}}:
+    1. If the result of running  IsDetachedBuffer (described in
+        [[!ECMASCRIPT]]) on |data| is `false`, return `false`.
+    2. If |data| is [=empty=], return `false`.
+4. If {{ImageDecoderInit/desiredWidth}} [=map/exists=] and
+    {{ImageDecoderInit/desiredHeight}} does not exist, return `false`.
+5. If {{ImageDecoderInit/desiredHeight}} [=map/exists=] and
+    {{ImageDecoderInit/desiredWidth}} does not exist, return `false`.
+6. Return `true`.
+
+A <dfn>valid image MIME type</dfn> is a string that is a [=valid MIME type
+string=] and for which the `type`, per Section 3.1.1.1 of [[RFC7231]], is
+`image`.
+
+: <dfn dict-member for=ImageDecoderInit>type</dfn>
+:: String containing the MIME type of the image file to be decoded.
+
+: <dfn dict-member for=ImageDecoderInit>data</dfn>
+:: {{BufferSource}} or {{ReadableStream}} of bytes representing an encoded
+    image file as described by {{ImageDecoderInit/type}}.
+
+: <dfn dict-member for=ImageDecoderInit>premultiplyAlpha</dfn>
+:: Controls whether decoded outputs' color channels are to be premultiplied by
+    their alpha channel, as defined by {{ImageBitmapOptions/premultiplyAlpha}}
+    in {{ImageBitmapOptions}}.
+
+: <dfn dict-member for=ImageDecoderInit>colorSpaceConversion</dfn>
+:: Controls whether decoded outputs' color space is converted or ignored, as
+    defined by {{ImageBitmapOptions/colorSpaceConversion}} in
+    {{ImageBitmapOptions}}.
+
+: <dfn dict-member for=ImageDecoderInit>desiredWidth</dfn>
+:: Indicates a desired width for decoded outputs. Implementation is best
+    effort; decoding to a desired width may not be supported by all formats/
+    decoders.
+
+: <dfn dict-member for=ImageDecoderInit>desiredHeight</dfn>
+:: Indicates a desired height for decoded outputs. Implementation is best
+    effort; decoding to a desired height may not be supported by all
+    formats/decoders.
+
+ImageDecodeResult Interface {#imagedecoderesult-interface}
+----------------------------------------------------------
+<pre class='idl'>
+<xmp>
+dictionary ImageDecodeResult {
+  required VideoFrame image;
+  required boolean complete;
+};
+</xmp>
+</pre>
+
+: <dfn dict-member for=ImageDecodeResult>image</dfn>
+:: The decoded image.
+
+: <dfn dict-member for=ImageDecodeResult>complete</dfn>
+:: Indicates whether {{ImageDecodeResult/image}} contains the final full-detail
+    output.
+
+    NOTE: {{ImageDecodeResult/complete}} is always `true` when
+        {{ImageDecoder/decode()}} is invoked with |completeFramesOnly| set to
+        `true`.
+
+ImageTrackList Interface {#imagetracklist-interface}
+----------------------------------------------------
+<pre class='idl'>
+<xmp>
+interface ImageTrackList : EventTarget {
+  getter ImageTrack (unsigned long index);
+
+  readonly attribute unsigned long length;
+  readonly attribute long selectedIndex;
+
```

Comment:
`readonly attribute ImageTrack? selected;` or possibly `current` to avoid weirdness like

`decoder.tracks.selected.selected = true;`